Added a section of Update parent POM page (#7194)
jenkins-infra/jenkins.io: content/doc/developer/tutorial-improve/update-parent-pom.adoc

// Create the branch
include::doc/developer/tutorial-improve/_create-a-branch.adoc[]

== Maven configuration

> Before updating the parent POM (Project Object Model) using the "mvn -ntp version:update-parent" command, you must make sure that the Maven configuration files are correctly configured.

Reviewer: ```suggestion
Before you proceed with the parent POM update, you must make sure that Maven is installed on your machine and that it is configured correctly for Jenkins plugin development.
```
The abbreviation POM is explained above, no need to expand it again. No need to duplicate the `mvn` command either.

> This is important because the plugin:maven-hpi-plugin[maven-hpi-plugin], used for developing Jenkins plugins, might require some additional global configurations to function correctly with older plugin environments.

Reviewer 2: ```suggestion
This is important because the link:https://github.com/jenkinsci/maven-hpi-plugin[maven-hpi-plugin], used for developing Jenkins plugins, requires some additional global configurations to function correctly with older plugin environments.
```
The plugin macro cannot be used here, as the maven-hpi-plugin is not a Jenkins plugins that can be found on plugins.jenkins.io. Therefore, it would need the link macro and have the link leading to the github page for the plugin.

Author: okay got it sir done

> If Maven reports that it cannot download artifacts during the update process, it might be due to improper configuration settings.

Reviewer 2: ```suggestion
Improper configuration settings can cause Maven to report that it cannot download artifacts during the update process.
```
Suggestion to reword the sentence and use active voice.

Author: yes sir done that

> To address this issue, create a file named `settings.xml` in the `/m2` directory, which is the default location for Maven settings, and configure it with the following content:

Reviewer 2: ```suggestion
To address this issue, create a file named `settings.xml` in the `.m2` directory, which is the default location for Maven settings, and configure it with the following content:
```
For the filepath, the folder should be `.m2` as opposed to `/m2`. The way this is currently written is misleading.

> [source,bash]

Reviewer 2: ```suggestion
[source,xml]
```
This code is written in xml, not bash

Author: oh sorry sir that was my error.
